kingpin/actors/aws/base.py

```diff
@@ -86,19 +90,23 @@ def __init__(self, *args, **kwargs):
         # By default, we will try to let Boto handle discovering its
         # credentials at instantiation time. This _can_ result in synchronous
         # API calls to the Metadata service, but those should be fast.
-        key = None
-        secret = None
-
+        #
         # In the event though that someone has explicitly set the AWS access
         # keys in the environment (either for the purposes of a unit test, or
         # because they wanted to), we use those values.
-        if aws_settings.AWS_ACCESS_KEY_ID and aws_settings.AWS_SECRET_ACCESS_KEY:
-            key = aws_settings.AWS_ACCESS_KEY_ID
-            secret = aws_settings.AWS_SECRET_ACCESS_KEY
+        #
+        # Note: these get defualted to None in aws_settings if they are not
+        # found which will tell boto3 to fallback to default behavior.
+        boto3_client_kwargs = {}
+        boto3_client_kwargs["aws_access_key_id"] = aws_settings.AWS_ACCESS_KEY_ID
+        boto3_client_kwargs[
+            "aws_secret_access_key"
+        ] = aws_settings.AWS_SECRET_ACCESS_KEY
+        boto3_client_kwargs["aws_session_token"] = aws_settings.AWS_SESSION_TOKEN
```

Comment:
@davidwhite-nd So it looks like this might only be valuable for unit testing by the looks of it. Other than that, it looks like we can completely remove this code.

But for now, I am just going to merge this PR. Alternatively, you can just unset the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` env vars before calling kingpin to get the same effect on older versions (as long as your keys are stored in a creds file.

Also, can you reverse-review this. I believe it is safe to pass in an unchecked dict as kwargs as the underlying boto3 library seems to handle None values gracefully (ie. reverts back to the default searching).